### PR TITLE
Add context menu command support

### PR DESCRIPTION
**Pull request by RafDevX**

### CODE REVIEW

**Comment by RafDevX** [comment]
Things to consider:
- whether to split `UserContextMenuCommands` and `MessageContextMenuCommands` from the get-go (vs. current solution of checking inside a unified handler); see https://github.com/ist-bot-team/ist-discord-bot/pull/100/files#diff-0c14a37a0833070d83b1840afa4f70d830ccf1dd45cfc2f69d783aaea9ae1c1aR37
- how to improve command logging format (audit, not debug) to look better / potentially include more information; see [`utils.stringifyContextMenuCommand`](https://github.com/ist-bot-team/ist-discord-bot/pull/100/files#diff-df34984041fefc50027ab0d898cb89d41c3c3716a9372c0a8dadcf9982bac56eR156)

Thoughts, @diogotcorreia?

**Comment by diogotcorreia** [comment]
@RafDevX 

- I'd split them. When registering the command, it can only be of one type (from what I've seen, if I'm wrong this does not apply then). Therefore, it'd make sense for the handler to be only of that type.
- Seems good to me as it is tbh

**Comment by RafDevX** [comment]
I've decided against splitting it as there would be no way to actually type-check on runtime.

I can tell TypeScript that the handler accepts a `MessageContextMenuInteraction` but, in the end, there's nothing stopping someone from registering both a Message and a User command with the same name* - which would be translated into the same handler (and one of them would error, expecting it the interaction to be of type X when it's of type Y).

As such, I think the best way to do this is to just use type guards inside handlers (through `isMessageContextMenuCommand` and `isUserContextMenuCommand`), as originally designed.

*as far as I can tell, Discord allows this (I just tested it)

File: src/bot.d.ts [hunk 1]
```diff
@@ -2,23 +2,44 @@
...
+	handler: InteractionHandler<I>;
 	permission?: CommandPermission;
 }
 
+export type CommandDescriptor =
+	| GenericCommandDescriptor<
+			SlashCommandBuilder,
+			Discord.ChatInputCommandInteraction
+	  >
+	// TODO: possibly split into Message and User here
```
Comment (diogotcorreia) [inline src/bot.d.ts:37]: Remove TODO

File: src/modules/utils.ts [hunk 2]
```diff
@@ -152,3 +152,30 @@ export function stringifyCommand(
...
+
+export function stringifyCommand(
+	interaction: Discord.CommandInteraction
+): string {
+	if (interaction.isChatInputCommand()) {
+		return stringifyChatInputCommand(interaction);
+	} else if (interaction.isContextMenuCommand()) {
+		return stringifyContextMenuCommand(interaction);
+	} else {
+		throw new Error("Unknown command type");
```
Comment (diogotcorreia) [inline src/modules/utils.ts:179]: Add a `logger.error({ interaction }, "Failed to stringify command due to unknown type)` here

**Review by diogotcorreia — CHANGES_REQUESTED** [review]

**Review by diogotcorreia — APPROVED** [review]

**Pull request merged**